### PR TITLE
Skeleton for git payload

### PR DESCRIPTION
**Pull request by indresh-28**
This is how in the git repository the changes looks like:
image

Customer has to go and create a PR

### CODE REVIEW

File: capten/model/server.gen.go [hunk 1]
```diff
@@ -91,6 +91,14 @@ type ApplicationDeployRequest struct {
 	OverrideValues string `json:"OverrideValues,omitempty"`
 }
 
+type ConfigureCICD struct {
+	PluginName string `json:"PluginName,omitempty"`
+	RepoURL    string `json:"RepoURL,omitempty"`
+	Token      string `json:"Token,omitempty"`
```
Comment (indresh-28) [inline capten/model/server.gen.go:97]: Don't store the creds, as it stored in the cassandra.
Read from the vault.

File: capten/model/server.gen.go [hunk 2]
```diff
@@ -91,6 +91,14 @@ type ApplicationDeployRequest struct {
 	OverrideValues string `json:"OverrideValues,omitempty"`
 }
 
+type ConfigureCICD struct {
+	PluginName string `json:"PluginName,omitempty"`
```
Comment (indresh-28) [inline capten/model/server.gen.go:95]: USe case, repourl, cred type(read from vault).

File: capten/config-worker/Dockerfile [hunk 3]
```diff
@@ -5,8 +5,9 @@ COPY ./ ./
...
-FROM scratch
-COPY --from=builder ./build/config_worker config_worker
+FROM alpine:3.16
+RUN apk add --no-cache libc6-compat
+COPY --from=builder ./build/config_worker /config_worker
+COPY --from=builder ./config-worker/git-config-files /git-config-files
 
-USER 65532:65532
-ENTRYPOINT ["./config_worker"]
+ENTRYPOINT ["/config_worker"]
```
Comment (indresh-28) [inline capten/config-worker/Dockerfile:13]: placeholder to replace the template configuration.

File: capten/common-pkg/plugins/git/git.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	_, err = w.Add(path)
+	if err != nil {
+		return err
+	}
+
+	_, err = w.Commit(msg, &git.CommitOptions{
+		Author: &object.Signature{
+			Name:  "capten-agent-bot",
```
Comment (indresh-28) [inline capten/common-pkg/plugins/git/git.go:53]: Take the env or use default.

File: capten/common-pkg/plugins/git/git.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	return nil
+}
+
+func (op *Operation) Push(branchName, token string) error {
+	return op.repository.Push(&git.PushOptions{RemoteName: "origin", Force: true,
+		Auth: &http.BasicAuth{
+			Username: "dummy", // yes, this can be anything except an empty string
+			Password: token,
+		},
+		RefSpecs: []config.RefSpec{config.RefSpec("refs/heads/main:refs/heads/" + branchName),
```
Comment (indresh-28) [inline capten/common-pkg/plugins/git/git.go:72]: Need confirmation. Add branch name to env.
Automatically create the PR after the push against the default repo branch

File: capten/config-worker/pkg/activities/git_configure.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+		return err
+	}
+
+	defer os.RemoveAll(dir) // clean up
+
+	if err := configPlugin.Clone(dir, params.RepoURL, params.Token); err != nil {
+		return err
+	}
+
+	cmd := exec.Command("cp", "--recursive", filepath.Join("./", GitTemplateDir, appDir), dir)
```
Comment (indresh-28) [inline capten/config-worker/pkg/activities/git_configure.go:59]: Look for alternative to remove the shell from the docker image. Try to use programmatically

File: capten/config-worker/Dockerfile [hunk 7]
```diff
@@ -6,7 +6,6 @@ RUN go mod download
 RUN CGO_ENABLED=0 go build -o ./build/config_worker config-worker/main.go
 
 FROM scratch
-COPY --from=builder ./build/config_worker config_worker
-
```
Comment (vramk23) [inline capten/config-worker/Dockerfile:10]: USER need to retained

File: capten/config-worker/Dockerfile [hunk 8]
```diff
@@ -6,7 +6,6 @@ RUN go mod download
...
-USER 65532:65532
-ENTRYPOINT ["./config_worker"]
+COPY --from=builder ./build/config_worker /config_worker
+COPY --from=builder ./config-worker/git-config-files /git-config-files
```
Comment (vramk23) [inline capten/config-worker/Dockerfile:10]: rename folder to /gitopts-template

under this folder different usecase to be used.

 /gitopts-templates
   /cicd
   /scm

for tekton use case, folder structure needed to be added under /cicd folder, not single file.

**Review by vramk23 — CHANGES_REQUESTED** [review]

**Review by vramk23 — APPROVED** [review]

**Pull request merged**